clustered/client/src/main/java/org/ehcache/clustered/client/internal/SimpleClusterTierManagerClientEntity.java

```diff
@@ -110,15 +89,22 @@ private EhcacheEntityResponse invokeInternal(TimeoutDuration timeLimit, EhcacheE
     return endpoint.beginInvoke().message(message).replicate(replicate).invoke();
   }
 
-  private static <T extends EntityResponse> T waitFor(TimeoutDuration timeLimit, InvokeFuture<T> future)
-      throws EntityException, TimeoutException {
+  /**
+   * Will wait forever on {@code Future.get()}. In case of interruption, it will take note and resume waiting on get. The
+   * interruption flag is then set if needed before returning the get value (or if an exception occurred).
+   *
+   * @param future Future we want to get
+   * @param <T> type of the response
+   * @return the result of the get
+   * @throws EntityException exception that might be thrown by the future in case of error
+   */
+  private static <T extends EntityResponse> T waitFor(InvokeFuture<T> future)
+      throws EntityException {
     boolean interrupted = false;
```

Comment:
Not your code but `boolean interrupted = Thread.interrupted()` is a better initialization here.

Reply:
Yep. Note that we also have `ExecutorUtil.waitFor` that is also initialized to `false`